tests/by_image/base-notebook/data/check_listening.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+import socket
+import time
+
+import requests
+
+
+def test_connect() -> None:
+    # Give some time for server to start
+    finish_time = time.time() + 10
+    sleep_time = 1
+    while time.time() < finish_time:
+        time.sleep(sleep_time)
+        try:
+            requests.get("http://localhost:8888/api")
+            break
+        except requests.RequestException:
+            pass
```

Comment:
There is a bug here - we fall through no matter if requests was successful or not, will fix it